### PR TITLE
LPS-181729 On portal startup make batch engine wait for activation of underlying ResourceImpl

### PR DESCRIPTION
**Pull request by matijapetanjek**
Hey @rotty3000, I had trouble making [this approach](https://github.com/liferay-headless/liferay-portal/pull/1172) work,  [componentFactory.newInstance()](https://github.com/liferay-headless/liferay-portal/pull/1172/commits/7c2be77d8b6a093958b9b833c46e598a43b10c67#diff-760781d7df7391517dd77dae0b22bf2b6ddd3728eb3a47e7489d642d6f1bbec1R149) wouldn't wait for activation of ResourceImpl(EntityResourceModel) service and instead would immediately fail with ComponentException.

Here is another attempt, influenced by your PR, at solving synchronization between ResourceImpls and batch engine on startup, if you have time please check if it makes sense.

Thank you!

### CODE REVIEW

**Comment by matijapetanjek** [comment]
ci:test:sf

**Comment by matijapetanjek** [comment]
To give more context, [this commit](https://github.com/liferay-headless/liferay-portal/pull/1196/commits/98926049cb8ec8c35744364919351f4437a34d37) contains the main change. I'm using serviceTracker to track particular ResourceImpl and start the ImportTask only once the ResourceImpl is registered.

**Comment by matijapetanjek** [comment]
ci:test:sf

**Comment by matijapetanjek** [comment]
ci:test:relevant

**Comment by matijapetanjek** [comment]
ci:reevaluate:1257401_624

File: modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/unit/BatchEngineUnitProcessorImpl.java [hunk 1]
```diff
@@ -81,6 +86,13 @@ public void processBatchEngineUnits(
 		}
 	}
 
+	@Activate
+	protected void activate(
+		BundleContext bundleContext, Map<String, Object> properties) {
```
Comment (ccorreagg) [inline modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/unit/BatchEngineUnitProcessorImpl.java:91]: The properties are not necessary

**Comment by rotty3000** [comment]
please hold off on more changes for the time being as I review the progress.

**Review by ccorreagg — COMMENTED** [review]
Just a short comment! Great PR @matijapetanjek

**Comment by 4lejandrito** [comment]
See https://github.com/liferay-headless/liferay-portal/pull/1341,

**Pull request closed without merging**